README.md

```diff
@@ -1,29 +1,280 @@
+----
+```
+
+As you can see, the macro takes care of the directories (`src/main/java` vs `src/main/groovy` vs `src/main/kotlin`) and the file extension.
+
+Following this same approach there are macros like:
+- `source`: Already explained.
+- `resource`: To include a file from the `src/main/resources` directory.
+- `test`: To include a file from the `src/main/test` directory. This macro also takes care of the suffix depending on the test framework. For example, with `test:GithubControllerTest[]` the macro will reference the file `GithubControllerTest.java` (or .kt) for Java and Kotlin and `GithubControllerSpec.groovy` for Groovy.
+- `testResource`: To include a file from the `src/main/test/resources` directory.
+- `callout`: To include a common callout snippet.
+
+In all the cases it is possible to pass additional parameters to the macros to customise them. For example, to extract a custom tag from a snippet, we can do `resource:application.yml[tag=githubconfig]`. Look for usages of those macros in the `guides` directory to find more examples.
+
+#### Special custom blocks
+
+There are also special custom blocks to exclude some code to be included in the generated guide based on some condition. This is useful when explaining something specific of the build tool (like how to run the tests with Gradle or Maven) or to exclude something depending on the language (for example do not render the GraalVM section in Groovy guides, as Groovy is not compatible with GraalVM).
+
+Example:
+
+```asciidoc
+:exclude-for-languages:kotlin
+<2> The Micronaut framework will not load the bean unless configuration properties are set.
+:exclude-for-languages:
+
+:exclude-for-languages:java,groovy
+<2> Kotlin doesn't support runtime repeatable annotations (see https://youtrack.jetbrains.com/issue/KT-12794[KT-12794]. We use a custom condition to enable the bean where appropriate.
+:exclude-for-languages:
+```
 
-## New Guide Template
+For Java and Groovy guides the first block will be included. For Kotlin guide, the second block will be included.
 
-To create a new guide create a new asciidoc file under `src/guides` with the following template: 
- 
+Example for build tool:
+
+```asciidoc
+:exclude-for-build:maven
+
+Now start the application. Execute the `./gradlew run` command, which will start the application on port 8080.
+
+:exclude-for-build:
+
+
+:exclude-for-build:gradle
+
+Now start the application. Execute the `./mvnw mn:run` command, which will start the application on port 8080.
+
+:exclude-for-build:
+```
+
+For a Gradle guide, the first block will be included. For a Maven guide, the second one will be included.
+
+As before, look for usages of the macro in the `guides` directory for more examples.
+
+### New Guide Template
+
+To create a new guide use the following template as the base asciidoc file:
 
 ```asciidoc
 include::{commondir}/common-header.adoc[]
```

Comment:
This needs to be updated from

`include::{commondir}/common-header.adoc[]`

to: `common:header.adoc[]`

see: https://github.com/micronaut-projects/micronaut-guides/pull/688